pkg/deployments/scaler.go

```diff
@@ -46,69 +70,53 @@ func (s *scaler) UpdateState(replicas, min, max int32) {
 // SetDesiredScale sets the desired scale of the scaler and scales
 // if needed.
 func (s *scaler) SetDesiredScale(n int32) {
-	log.Printf("SetDesiredScale(%v)", n)
-	s.compareScales(-1, s.applyMinMax(n))
-}
-
-func (s *scaler) applyMinMax(n int32) int32 {
-	s.mtx.Lock()
-	min := s.minScale
-	max := s.maxScale
-	s.mtx.Unlock()
-	if n < min {
-		n = min
-	} else if n > max {
-		n = max
-	}
-	return n
-}
-
-func (s *scaler) compareScales(current, desired int32) {
+	log.Printf("SetDesiredScale(%v), current: %v, min: %v, max: %v", n, s.currentScale, s.minScale, s.maxScale)
+	nMinMax := s.applyMinMax(n)
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
-
-	if current != -1 {
-		s.currentScale = current
-	}
-	if desired != -1 {
-		s.desiredScale = desired
-	}
-
-	if s.currentScale == -1 || s.desiredScale == -1 {
-		// Nothing to compare if we only have partial information
-		return
-	}
-
-	log.Printf("Comparing scales, current: %v, desired: %v", s.currentScale, s.desiredScale)
+	s.desiredScale = nMinMax
 
 	if s.desiredScale > s.currentScale {
 		// Scale up immediately.
-		go s.scaleFunc(s.desiredScale, false)
-		s.scaleDownStarted = false
+		log.Printf("Scaling up immediately")
+		go func() {
+			if err := s.scaleFunc(s.desiredScale, false); err != nil {
```

Comment:
Idiomatic way of doing this is to only check error condition, and return early on error instead of `if ... else`

Reply:
fixed, agree that does seem a bit cleaner